Internal/EnumStrings.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Reflection;
+using Silksong.ModMenu.Models;
+
+namespace Silksong.ModMenu.Internal;
+
+/// <summary>
+/// Utility for navigating enums through reflection, retrieving a map from values -> string representations.
+/// Respects ModMenu attributes.
+/// </summary>
+internal static class EnumStrings
+{
+    private static readonly Dictionary<Type, ReadOnlyDictionary<object, string>> enumStrings = [];
+
+    internal static ReadOnlyDictionary<object, string> ForType(Type enumType)
+    {
+        if (!enumType.IsEnum)
+            throw new ArgumentException($"{enumType.Name} is not an Enumeration");
+        if (enumStrings.TryGetValue(enumType, out var dict))
+            return dict;
+
+        Dictionary<object, List<FieldInfo>> fieldInfos = [];
+        foreach (var field in enumType.GetFields(BindingFlags.Public | BindingFlags.Static))
+        {
+            var rawValue = field.GetRawConstantValue();
+            if (fieldInfos.TryGetValue(rawValue, out var list))
+                list.Add(field);
+            else
+                fieldInfos.Add(rawValue, [field]);
+        }
+
+        Dictionary<object, string> names = [];
+        foreach (var value in enumType.GetEnumValues())
+        {
+            var rawValue = Convert.ChangeType(value, enumType.GetEnumUnderlyingType());
+            if (!fieldInfos.TryGetValue(rawValue, out var list))
+                continue;
+
+            bool ignore = false;
+            string? fieldName = null;
+            string? customName = null;
+            foreach (var field in list)
+            {
+                if (field.GetCustomAttribute<ModMenuIgnore>() != null)
+                {
+                    ignore = true;
```

Comment:
Is this expected behaviour? My understanding of this function is that you're doing a bunch of special handling for the case that they give multiple names to the same enum value, e.g. `enum MyEnum{Name1 = 1, Name2 = 1}`. If they put ModMenuIgnore on Name1 but not Name2, I think I would have expected Name2 to appear rather than skipping the value entirely.